zeebe/dynamic-config/src/test/java/io/camunda/zeebe/dynamic/config/api/ClusterPurgeRequestTransformerTest.java

```diff
@@ -0,0 +1,190 @@
+    // given
+    final var transformer = new PurgeRequestTransformer();
+
+    final ClusterConfiguration currentTopology =
+        ClusterConfiguration.init()
+            .addMember(id0, MemberState.initializeAsActive(Map.of()))
+            .addMember(id1, MemberState.initializeAsActive(Map.of()))
+            .updateMember(id0, m -> m.addPartition(0, PartitionState.active(2, partitionConfig)))
+            .updateMember(id0, m -> m.addPartition(1, PartitionState.active(1, partitionConfig)))
+            .updateMember(id1, m -> m.addPartition(0, PartitionState.active(1, partitionConfig)))
+            .updateMember(id1, m -> m.addPartition(1, PartitionState.active(2, partitionConfig)));
+
+    // when
+    final var result = transformer.operations(currentTopology);
+
+    // then
+    assertThat(result)
+        .isRight()
+        .right()
+        .satisfies(
+            operations -> {
+              assertThat(operations).hasSize(9);
+              assertThat(operations)
+                  .containsExactly(
+                      new PartitionLeaveOperation(id0, 0, true),
+                      new PartitionLeaveOperation(id0, 1, true),
+                      new PartitionLeaveOperation(id1, 0, true),
+                      new PartitionLeaveOperation(id1, 1, true),
+                      new DeleteHistoryOperation(id0),
+                      new PartitionBootstrapOperation(id0, 0, 2, Optional.of(partitionConfig)),
+                      new PartitionBootstrapOperation(id1, 1, 2, Optional.of(partitionConfig)),
+                      new PartitionJoinOperation(id1, 0, 1),
+                      new PartitionJoinOperation(id0, 1, 1));
+            });
+  }
+
+  @Test
+  void purgeClusterShouldBootstrapPartitionsInOrder() {
+    // given
+    final var transformer = new PurgeRequestTransformer();
+
+    final ClusterConfiguration currentTopology =
+        ClusterConfiguration.init()
+            .addMember(id0, MemberState.initializeAsActive(Map.of()))
+            .addMember(id1, MemberState.initializeAsActive(Map.of()))
+            .updateMember(id0, m -> m.addPartition(0, PartitionState.active(2, partitionConfig)))
+            .updateMember(id0, m -> m.addPartition(1, PartitionState.active(1, partitionConfig)))
+            .updateMember(id0, m -> m.addPartition(2, PartitionState.active(2, partitionConfig)))
+            .updateMember(id1, m -> m.addPartition(0, PartitionState.active(1, partitionConfig)))
+            .updateMember(id1, m -> m.addPartition(1, PartitionState.active(2, partitionConfig)))
+            .updateMember(id1, m -> m.addPartition(2, PartitionState.active(1, partitionConfig)));
+
+    // when
+    final var result = transformer.operations(currentTopology);
+
+    // then
+    assertThat(result)
+        .isRight()
+        .right()
+        .satisfies(
+            operations -> {
+              assertThat(operations).hasSize(13);
+              assertThat(operations)
+                  .containsExactly(
```

Comment:
:wrench: I think there's a assert method to check if it contains a sequence, so you don't need to specify _all_ operations, but instead you can specify just the sequence you're interested in (in this case, the bootstrap).